src/dmd/backend/evalu8.d

```diff
@@ -275,33 +282,37 @@ int iffalse(elem *e)
  */
 
 elem * evalu8(elem *e, goal_t goal)
-{   elem *e1,*e2;
+{
+    elem* e1;
+    elem* e2;
     tym_t tym,tym2,uns;
-    unsigned op;
+    uint op;
     targ_int i1,i2;
     targ_llong l1,l2;
     targ_ldouble d1,d2;
-    elem esave;
+    elem esave = void;
+
+    static bool unordered(T)(T d1, T d2) { return isnan(d1) || isnan(d2); }
 
     assert((statusFE() & 0x3800) == 0);
-    assert(e && EOP(e));
-    op = e->Eoper;
+    assert(e && !OTleaf(e.Eoper));
```

Comment:
Why not define EOP or EBIN as a function?

Reply:
I don't think those macros paid off.